### PR TITLE
VZ-5630 Add install/uninstall of kube-state-metrics to Verrazzano

### PR DESCRIPTION
**Pull request by desagar**
# Description

Please include a summary of the change and which issue is fixed.  If there are any dependencies, for example on a PR in another repository, please list those.

Fixes VZ-9999

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/registry/registry_test.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ package registry
 import (
 	"testing"
 
+	"github.com/verrazzano/verrazzano/platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics"
 	"github.com/verrazzano/verrazzano/platform-operator/controllers/verrazzano/component/weblogic"
 
 	"github.com/stretchr/testify/assert"
```
Comment (davisgomes) [inline platform-operator/controllers/verrazzano/component/registry/registry_test.go:12]: These imports need reformatting if you get around to it.

File: platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	"github.com/verrazzano/verrazzano/platform-operator/internal/k8s/status"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+import (
+	"context"
+)
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics.go:19]: Combine the imports?

File: platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+			SupportsOperatorInstall: true,
+			MinVerrazzanoVersion:    constants.VerrazzanoVersion1_3_0,
+			ImagePullSecretKeyname:  "imagePullSecrets[0].name",
+			ValuesFile:              filepath.Join(config.GetHelmOverridesDir(), "kube-state-metrics-values.yaml"),
+			Dependencies:            []string{},
+		},
+	}
+}
+
+// IsEnabled returns true if the Prometheus Operator is enabled or if the component is not specified
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component.go:48]: Copy/paste: Operator

File: platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+// in the Verrazzano CR.
+func (c kubeStateMetricsComponent) IsEnabled(effectiveCR *vzapi.Verrazzano) bool {
+	comp := effectiveCR.Spec.Components.KubeStateMetrics
+	if comp == nil || comp.Enabled == nil {
+		return true
+	}
+	return *comp.Enabled
+}
+
+// IsReady checks if the Prometheus Operator deployment is ready
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component.go:58]: Copy/paste

File: platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+func (c kubeStateMetricsComponent) IsReady(ctx spi.ComponentContext) bool {
+	if c.HelmComponent.IsReady(ctx) {
+		return isDeploymentReady(ctx)
+	}
+	return false
+}
+
+// PreInstall updates resources necessary for the Prometheus Operator Component installation
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component.go:66]: Copy/paste

File: platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component_test.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	vzapi "github.com/verrazzano/verrazzano/platform-operator/apis/verrazzano/v1alpha1"
+	"github.com/verrazzano/verrazzano/platform-operator/controllers/verrazzano/component/spi"
+)
+
+const profilesRelativePath = "../../../../../manifests/profiles"
+
+// TestIsEnabled tests the IsEnabled function for the Prometheus Operator component
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component_test.go:16]: Copy/paste

File: platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component_test.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	falseValue := false
+	trueValue := true
+	tests := []struct {
+		name       string
+		actualCR   vzapi.Verrazzano
+		expectTrue bool
+	}{
+		{
+			// GIVEN a default Verrazzano custom resource
+			// WHEN we call IsReady on the Prometheus Operator component
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component_test.go:27]: Copy/paste

**Review by fredtibbitts — APPROVED** [review]
Looks good, just a handful of minor comments.

File: platform-operator/controllers/verrazzano/component/registry/registry_test.go (hunk 1 above)
Comment (desagar) [reply platform-operator/controllers/verrazzano/component/registry/registry_test.go]: done

**Review by davisgomes — APPROVED** [review]
looks good

**Review by fredtibbitts — APPROVED** [review]

**Review by stevshap — APPROVED** [review]

**Pull request merged**